specs/vep-2272-complete-data-job-configuration-persistence/README.md

```diff
@@ -95,75 +95,111 @@ There would be no changes to the public APIs.
 
 ## Detailed design
 
-### Database data model changes
+### Kubernetes Cron Jobs Synchronizer
+
+To enhance data integrity and synchronization between a database and Kubernetes, an asynchronous process will be
+implemented. This process will leverage Spring Scheduler and Scheduler Lock, similar to the
+[DataJobMonitorCron.watchJobs()](https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMonitorCron.java#L84).
+
+The overall purpose of this process is to iterate through the data job deployment configurations stored in the database
```

Comment:
Have you looked into Kubernetes Informer ([link to example](https://github.com/Netflix-Skunkworks/kubernetes-client-java/blob/master/examples/src/main/java/io/kubernetes/client/examples/InformerExample.java) )

Apparently it's a higher level abstraction than watch and handles re-tries, disconnections, caching. 

It might even solve some of the issues with watch we have for all I know.

Reply:
yes, you are correct. Got confused here . the SYnc is in the other direction.